bullet_train-obfuscates_id/app/models/concerns/obfuscates_id.rb

```diff
@@ -23,7 +23,11 @@ def decode_id(id)
     end
 
     def find(*ids)
-      super(*ids.map { |id| decode_id(id) })
+      if respond_to?(:slug_attribute)
+        send("find_by_#{slug_attribute}".to_sym, ids.first)
```

Comment:
Okay, it looks like the failure when trying to update the record is happening here.

Say for instance we try to save the record with a restricted path, `admin`.

The validator will raise an error, but when returning to the edit page, it tries to find it via the faulty id:

```
    25: def find(*ids)
 => 26:   binding.pry
    27:   if respond_to?(:slug_attribute)
    28:     send("find_by_#{slug_attribute}".to_sym, ids.first)
    29:   else
    30:     super(*ids.map { |id| decode_id(id) })
    31:   end
    32: end

[1] pry(Project)> ids
=> ["admin"]
[2] pry(Project)> Project.first
  Project Load (1.6ms)  SELECT "projects".* FROM "projects" ORDER BY "projects"."id" ASC LIMIT $1  [["LIMIT", 1]]
  ↳ (pry):2:in `find'
=> #<Project:0x00007f12e88daf20
 id: 5,
 team_id: 1,
 title: "Test",
 foo: "example-project",
 created_at: Mon, 05 Feb 2024 10:25:23.651803000 UTC +00:00,
 updated_at: Mon, 05 Feb 2024 10:25:23.651803000 UTC +00:00>
```

The faulty id hasn't persisted to the database, and here I'm trying to update the record with a valid slug, so I'm not sure why `admin` is still there. Will have to do some more investigating.

Reply:
Okay, that was a pretty tough one. Fixed it in https://github.com/bullet-train-co/bullet_train-core/pull/764/commits/e3a2a644cb66ceca20801c9e2ce78b9204750c73.